src/libcore/future/pending.rs

```diff
@@ -0,0 +1,57 @@
+use crate::future::Future;
+use crate::marker;
+use crate::pin::Pin;
+use crate::task::{Context, Poll};
+
+/// Creates a future which never resolves, representing a computation that never
+/// finishes.
+///
+/// This `struct` is created by the [`pending`] function. See its
+/// documentation for more.
+///
+/// [`pending`]: fn.pending.html
+#[unstable(feature = "future_readiness_fns", issue = "70921")]
+#[derive(Debug)]
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+pub struct Pending<T> {
+    _data: marker::PhantomData<T>,
+}
+
+/// Creates a future which never resolves, representing a computation that never
+/// finishes.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(future_readiness_fns)]
+/// use core::future;
+///
+/// # async fn run() {
+/// let future = future::pending();
+/// let () = future.await;
+/// unreachable!();
+/// # }
+/// ```
+#[unstable(feature = "future_readiness_fns", issue = "70921")]
+pub fn pending<T>() -> Pending<T> {
```

Comment:
I know we already see this pattern elsewhere in std::io, but it's generally not the standard idom for constructors. Maybe these should just be `impl<T> Pending<T> { pub fn new() -> Self { ... } }`?

Reply:
Similar methods also exist in the `iter` submodule in the form of `iter::{never, once, repeat}`.

The functions proposed in this PR feel similar to their counterparts in `iter` — both sets are convenience functions, and having them exist as shorthands under the submodule makes them the most convenient.

Reply:
Yeah, I guess it's best to follow those conventions here then.